_posts/2020-02-14-where-are-we-at-with-waku.md

```diff
@@ -0,0 +1,147 @@
+
+For more details on these bottlenecks, uncertainty and mitigations, see [Scalability estimate: How many users can Waku and the Status app support?](https://discuss.status.im/t/scalability-estimate-how-many-users-can-waku-and-the-status-app-support/1514).
+
+TODO: Elaborate on bottleneck 3, kad etc
+
+## Simulation
+
+The ultimate test is real-world usage. Until then, we have a simulation thanks to Kim De Mey from the Nimbus team!
+
+![](assets/img/waku_simulation.jpeg)
+
+We have two network topologies, Star and full mesh, with 6 randomly connected nodes, one traditional light node with bloom filter (Whisper style) and one Waku light node.
+
+One of the full nodes sends 1 envelope over 1 of the 100 topics that the two light nodes subscribe to. After that, it sends 10000 envelopes over random topics.
+
+For light node, bloom filter is set to almost 10% false positive (bloom filter: n=100, k=3, m=512). It shows the number of valid and invalid envelopes received for the different nodes.
+
+**Star network:**
+
+| Description     | Peers | Valid | Invalid |
+|-----------------|-------|-------|---------|
+| Master node     |     7 | 10001 |       0 |
+| Full node 1     |     3 | 10001 |       0 |
+| Full node 2     |     1 | 10001 |       0 |
+| Full node 3     |     1 | 10001 |       0 |
+| Full node 4     |     1 | 10001 |       0 |
+| Full node 5     |     1 | 10001 |       0 |
+| Light node      |     2 |   815 |       0 |
+| Waku light node |     2 |     1 |       0 |
+
+**Full mesh:**
+
+| Description     | Peers | Valid | Invalid |
+|-----------------|-------|-------|---------|
+| Full node 0     |     7 | 10001 |   20676 |
+| Full node 1     |     7 | 10001 |    9554 |
+| Full node 2     |     5 | 10001 |   23304 |
+| Full node 3     |     5 | 10001 |   11983 |
+| Full node 4     |     5 | 10001 |   24425 |
+| Full node 5     |     5 | 10001 |   23472 |
+| Light node      |     2 |   803 |     803 |
+| Waku light node |     2 |     1 |       1 |
+
+Things to note:
+- Whisper light node with ~10% false positive gets ~10% of total traffic
+- Waku light node gets ~1000x less envelopes than Whisper light node
+- Full mesh results in a lot more duplicate messages, expect for Waku light node
+
+Run the simulation yourself [here](https://github.com/status-im/nimbus/tree/master/waku#testing-waku-protocol). The parameters are configurable, and it is integrated with Prometheus and Grafana.
+
+## Difference between Waku and Whisper
+
+Summary of main differences between Waku v0 spec and Whisper v6, as described in [EIP-627](https://eips.ethereum.org/EIPS/eip-627):
+
+- Handshake/Status message not compatible with shh/6 nodes; specifying options as association list
+- Include topic-interest in Status handshake
+- Upgradability policy
+- `topic-interest` packet code
```

Comment:
There has been some developments around it to remove it and replace with a packet to exchange `statusOptions` like in the handshake.